### ISSUE TITLE
Missing default General subcategory meter due to the use of (valid) End Use category aliases

### ISSUE THREAD

**Issue by jcyuan2020** [NREL/EnergyPlus#10224]
Issue overview
--------------

While working on PR10011, there is another issue found in the code, in that some of the non-standard “alias” end use category might accidently skip adding a `General` subcategory meter since the end use category name “standardization” comes in too late.

The affected code is here:
https://github.com/NREL/EnergyPlus/blob/8134ee039ce1cba8a9b53dba5e3e006e945d7560/src/EnergyPlus/OutputProcessor.cc#L4440-L4445

If “EndUse” is a non-standard alias, it would normally be converted to a “standard” name when a meter is added. However, at this point in setting up variables, the conversion has not happened yet. This would then cause a look up failure and then later a missing meter with “General” sub category.

One item that was affected it the “DHW” alias used in the code---which was trying to be addressed in PR10011 to eliminate the use of that alias. However, since there are so many other alias “end use” that could be used according to the “standardization” code section, it is likely that the other ones could be affected as well.

The solution is to do a refactoring so that “standardization” portion could happen earlier, or the above affected code portion could occur later. Code changes will follow soon.

### Details
Some additional details for this issue (if relevant):
 - Platform (Operating system, version): All
 - Version of EnergyPlus (if using an intermediate build, include SHA): v23.2 pre-release, and possiblly in earlier versions as well
 - Unmethours link or helpdesk ticket number

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Defect file added (list location of defect file here)
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

**Comment by mjwitte** [comment]
Even better would be to make this parameter an `enum`, but that's a big lift, touching all 4k+ calls to `SetupOutputVariable`.

**Comment by jcyuan2020** [comment]
> Even better would be to make this parameter an `enum`, but that's a big lift, touching all 4k+ calls to `SetupOutputVariable`.

@mjwitte I agree that should be the way to go which should give more rigorous checks and actually would save many line of check code later. I would be glad to put this as a next step of work. There are few more string parameters such as reporting frequency, resource type that can also benefit from this method the same way.

Actually, to explore this a little bit, I've written an overload wrapper function to allow what the new ways to call `SetOuputVariable` with a few `enum` alternatives. This will enable the new way to call `SetOuputVariable`, but still keeps the old ways to call unaffected (thousands of instances) during the transition. Will posted in a new PR about this soon.

**Comment by dareumnam** [comment]
This defect appears to have been resolved by #10347 (OutputProcessor Refactor). That PR absorbed the fix from #10228 (which was labeled Fixes this issue) into the new framework, converting the `EndUse` argument of `SetupOutputVariable` to the `EndUseCat` enum and introducing `standardizeEndUseSub()`. Because the enum mapping (`endUseCat2endUse[]`) always resolves to the correct `EndUse` regardless of whether a non-standard alias was used, the missing "General" sub-category meter no longer occurs. #10228 was closed in favor of #10347, but since #10347 did not include an auto-close keyword for this issue, it was left open. I believe this can now be closed.
Happy to add a unit test regarding this issue, please let me know.
